pyradiosky/skymodel.py

```diff
@@ -412,8 +416,14 @@ def __init__(
                 freq_array = freqs_use
                 reference_frequency = None
 
-        if nside is not None:
+        if component_type is not None:
+            self._set_component_type_params(component_type)
+        elif nside is not None:
             self._set_component_type_params("healpix")
+        else:
+            self._set_component_type_params("point")
```

Comment:
This all looks fine, but I did have a thought or two when looking at the actual function `_set_component_type_params`:
- The set function is defined below this one, that does violate our desire to have dependents above functions but can easily be moved
- The set function does not seem to do any real validation of input, it either checks for equality to "healpix" or defaults to point. Is this the behavior we want especially if we're allowing a new user input to be passed directly to that function?

Reply:
good points. I added this part later, I'll move it and do some better checking (this would get caught in the `check` function but that can be turned off).